src/aws.ts

```diff
@@ -504,6 +507,237 @@ export namespace $AWS {
       },
     });
   }
+
+  export const SDK: StepFunctionSDK = new Proxy<any>(
```

Comment:
nit: If we support lambda, `StepFunctionSDK` is probably not the best name.

Reply:
Yeah, I was wondering if we should update $AWS be this proxy and then have separate handlers for type safe ones?

Reply:
I'd be ok with that. for example putting them as static methods on their corresponding resource objects.

`Table.aws.putItem(...)`
`StepFunction.aws.startExecution()`

They would be more discoverable as well.

Then we'd have 3 forms:

1. Generalized API calls inferred from the SDK or generated using smithy (`$AWS.DynamoDb.putItem({ TableName: myTable.tableName, ... })`)
   1. Brute force access to the SDK for coverage
3. Special call that accepts a construct at the resource level, implemented per resource (`Table.aws.putItem()` or `Table.putItem({ Table: myTable, ... })`)
   1. gives full access to the SDK, most accurate at permission inference and specific integrations. created by hand (? for now?)
5. Specific resource which has invokable methods or is invokable (`myTable.putItem`)
   1. does not give full access to the SDK input or response, cleanest, created by hand

Reply:
@thantos good catch, I renamed it to just `SDK` now 👍 

If I understand you two correctly you are thinking about making `$AWS` what is currently `$AWS.SDK` and move all manually implemented calls somewhere else?

Reply:
Yeah it might be the right approach - first go broad across all services, then build up from that with targeted type-safe (data aware) resource implementations. `$AWS` should map 1:1 with the aws-sdk as the name is inspired by this pattern:

```ts
import AWS from "aws-sdk"
```

+1 to @thantos's approach of moving the type-safe APIs to the Resource's Class as static methods, e.g. `Table.BatchGetItem( .. )`.

Reply:
Should we do it as a part of this change, or merge it in with a different name and then follow-up with a migration?

Reply:
I would prefer getting this in first under `$AWS.SDK` and then move everything as I am not too sure I can do the migration with my limited knowledge of the project.

Reply:
Makes sense! I agree.

Reply:
This is already an impactful change worth merging, we can easily make a breaking change later in a refactor. Thanks again for putting this together.

Reply:
100% agree.